src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -28,74 +28,88 @@ internal class CookieAuthenticationHandler : AuthenticationHandler<CookieAuthent
         private DateTimeOffset? _renewExpiresUtc;
         private string _sessionKey;
 
-        protected override async Task<AuthenticationTicket> AuthenticateAsync()
+        private async Task<AuthenticationTicket> ConfigureTicket()
```

Comment:
RetrieveTicketAsync? Also, is there any point in having this run more than once? RetrieveTicketOnceAsync?